test/corpus/types

```diff
@@ -49,3 +49,47 @@ Address
     (expr
       (constant
         (ipv6)))))
+
+================================================================================
+Enum with deprecated
+================================================================================
+
+type enum_with_deprecated: enum {
+    ZERO = 0,
+    ONE = 1 &deprecated,
+    TWO = 2 &deprecated="two is deprecated :(",
+    THREE = 3,
+};
+--------------------------------------------------------------------------------
```

Comment:
Nit: This tests enum labels with value, and with value and attribute (both without and with attribute value). WDYT about adding all permissible variations?

```suggestion
================================================================================
Enum
================================================================================

type E: enum {
    ZERO = 0,
    ONE = 1 &deprecated,
    TWO = 2 &deprecated="two is deprecated :(",
    THREE &deprecated,
    FOUR &deprecated="two is deprecated :(",
    FIVE,
};
--------------------------------------------------------------------------------
```

Reply:
sure, added it in a separate decl so it's valid code outside of tree sitter